gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/impl/sql/GenericStatement.java

```diff
@@ -138,33 +138,33 @@ Licensed to the Apache Software Foundation (ASF) under one or more
             "^\\s*\\{?\\s*(DROP|TRUNCATE)\\s+(TABLE|INDEX)\\s+",
             Pattern.CASE_INSENSITIVE);
 	public static final Pattern DELETE_STMT = Pattern.compile(
-            "^\\s*\\{?\\s*DELETE\\s+FROM\\s+.*", Pattern.CASE_INSENSITIVE);
+            "^\\s*\\{?\\s*DELETE\\s+FROM\\s+", Pattern.CASE_INSENSITIVE);
         private static final Pattern ignoreStmts = Pattern.compile(
             ("\\s.*(\"SYSSTAT\"|SYS.\")"), Pattern.CASE_INSENSITIVE | Pattern.DOTALL);
         //private ProcedureProxy procProxy;
         private final GfxdHeapThresholdListener thresholdListener;
         private THashMap ncjMetaData = null;
         private static final Pattern STREAMING_DDL_PREFIX =
-            Pattern.compile("\\s*STREAMING\\s+.*",
+            Pattern.compile("^\\s*STREAMING\\s+",
                 Pattern.CASE_INSENSITIVE | Pattern.DOTALL);
-        private static final Pattern INSERT_INTO_TABLE_SELECT_PATTERN =
-            Pattern.compile(".*INSERT\\s+INTO\\s+(TABLE)?.*\\s+SELECT\\s+.*",
+        private static final Pattern INSERT_OR_PUT_INTO_TABLE_SELECT_PATTERN =
+            Pattern.compile("^\\s*(INSERT|PUT)\\s+INTO\\s+(TABLE)?.*\\s+SELECT\\s+",
                 Pattern.CASE_INSENSITIVE | Pattern.DOTALL);
         private static final Pattern DML_TABLE_PATTERN =
-            Pattern.compile("^\\s*(INSERT|UPDATE|DELETE)\\s+.*",
+            Pattern.compile("^\\s*(INSERT|UPDATE|DELETE|PUT)\\s+",
                 Pattern.CASE_INSENSITIVE | Pattern.DOTALL);
-        private static final Pattern PUT_INTO_TABLE_SELECT_PATTERN =
-            Pattern.compile(".*PUT\\s+INTO\\s+(TABLE)?.*\\s+SELECT\\s+.*",
+        private static final Pattern NON_ROUTED_QUERY =
+            Pattern.compile("^\\s*\\{?\\s*(CALL|EXECUTE)\\s+",
                 Pattern.CASE_INSENSITIVE | Pattern.DOTALL);
         private static final Pattern FUNCTION_DDL_PREFIX =
-            Pattern.compile("\\s?(CREATE|DROP)\\s+FUNCTION\\s+.*",
+            Pattern.compile("^\\s?(CREATE|DROP)\\s+FUNCTION\\s+",
```

Comment:
If there is more than 1 space before CREATE or DROP then this will not match. This mistake is there in the previous code as well. It should be \\s*(CREATE|DROP).
Isn't it?

Reply:
Yes, will correct.